File: packages/contracts/src/billing-customer.ts
import type { EntityId, VersionedRecord } from "./common";

export type BillingCustomerType = "company" | "individual";

export interface BillingCustomerRecord extends VersionedRecord {
  organization_id: EntityId;
  name: string;
  customer_type: BillingCustomerType;
  email: string | null;
  phone: string | null;
  address: string | null;
  notes: string | null;
}

export interface BillingCustomerCreateRequest {
  name: string;
  customer_type: BillingCustomerType;
  email?: string | null;
  phone?: string | null;
  address?: string | null;
  notes?: string | null;
}

export interface BillingCustomerUpdateRequest {
  name?: string | null;
  customer_type?: BillingCustomerType | null;
  email?: string | null;
  phone?: string | null;
  address?: string | null;
  notes?: string | null;
}

File: apps/api/app/schemas/billing_customer.py
from uuid import UUID

from pydantic import BaseModel, Field

from app.db.models.billing_customer import BillingCustomerType
from app.schemas.common import BaseReadModel


class BillingCustomerRead(BaseReadModel):
    organization_id: UUID
    name: str
    customer_type: BillingCustomerType
    email: str | None
    phone: str | None
    address: str | None
    notes: str | None


class BillingCustomerCreateRequest(BaseModel):
    name: str = Field(min_length=2, max_length=160)
    customer_type: BillingCustomerType = BillingCustomerType.COMPANY
    email: str | None = Field(default=None, max_length=160)
    phone: str | None = Field(default=None, max_length=32)
    address: str | None = Field(default=None, max_length=2000)
    notes: str | None = Field(default=None, max_length=2000)


class BillingCustomerUpdateRequest(BaseModel):
    name: str | None = Field(default=None, min_length=2, max_length=160)
    customer_type: BillingCustomerType | None = None
    email: str | None = Field(default=None, max_length=160)
    phone: str | None = Field(default=None, max_length=32)
    address: str | None = Field(default=None, max_length=2000)
    notes: str | None = Field(default=None, max_length=2000)
